lib/Template/Constants.pm

```diff
@@ -121,13 +121,12 @@ sub debug_flags {
     my (@flags, $flag, $value);
     $debug = $self unless defined($debug) || ref($self);
     
-    if ($debug =~ /^\d+$/) {
+    if (length $debug && $debug !~ tr{0-9}{}c) {
```

Comment:
hard to read but correct

Reply:
but the length check isn't done in the original code. why do it here?